include/manager.hpp

```diff
@@ -80,30 +80,20 @@ class Manager
      *
      * API can be used to read VPD keyword.
      *
-     * To read a keyword from cache, input path should be D-bus object path.
-     * Eg: ("/xyz/openbmc_project/inventory/system/chassis/motherboard").
-     *
-     * To read a keyword from hardware, input path should be EEPROM path.
-     * Eg: ("/sys/bus/i2c/drivers/at24/9-0052/eeprom").
-     *
-     * To read keyword from IPZ type VPD, input data should be in the form of
+     * To read keyword of type IPZ, input data should be in the form of
      * (Record, Keyword). Eg: ("VINI", "SN").
      *
      * To read keyword from keyword type VPD, just keyword name has to be
      * supplied as input. Eg: ("SN").
      *
-     * The target to read can either be cache/hardware, whose values are 0/1
-     * respectively.
-     *
      * @param[in] i_path - D-bus object path/EEPROM path.
-     * @param[in] i_data - Data to be read.
-     * @param[in] i_target - Target location to read from (0/1).
+     * @param[in] i_paramsToReadData - Input required to perform read.
      *
-     * @return Read value in array of bytes.
+     * @return Read value in variant of array of bytes.
      */
-    types::BinaryVector readKeyword(const types::Path i_path,
-                                    const types::VpdData i_data,
-                                    const uint8_t i_target);
+    types::DbusVariantType
```

Comment:
Are you supporting partial DBus paths as input to the API?
eg: "/system/chassis/motherboard"

If yes, why?

Reply:
Then the implementation becomes specific to how we design our JSON.
There is no rule that JSON should have partial paths. Where as if user is supposed to pass Dbus path, should pass a valid DBus path.